server/embed/config_tracing.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package embed
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpgrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
+	"go.uber.org/zap"
+)
+
+const maxSamplingRatePerMillion = 1000000
+
+func (e *Etcd) setupTracingExporter(ctx context.Context) (exporter tracesdk.SpanExporter, options []otelgrpc.Option, err error) {
```

Comment:
Can change this to method that takes ServerConfig as argument. This function doesn't need anything from embed.Etcd than e.cfg. As so we should try to keep is separate from embed.Etcd [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) and make them separate. 

Best even move it to separate package `server/embed/tracing` this way we can avoid this function reading embed.Etcd private fields and functions.

Reply:
I followed the pattern we have with logging here, where we have config_logging.go. But will change that takes config instead, good suggestion!

Reply:
Ok, good point. I also prefer consistent quality, instead of having each module written differently. 
But this is a good candiate for a follow up PR. I would be happy to help with that.